src/Sarif.Multitool/Rules/SARIF2004.OptimizeFileSize.cs

```diff
@@ -57,6 +73,58 @@ protected override void Analyze(Run run, string runPointer)
             AnalyzeIdOnlyRules(run, runPointer);
         }
 
+        protected override void Analyze(Result result, string resultPointer)
+        {
+            ReportUnnecessaryAnalysisTarget(result, resultPointer);
+            ReportRuleDuplication(result, resultPointer);
+        }
+
+        private void ReportRuleDuplication(Result result, string resultPointer)
+        {
+            if (result.Rule != null)
+            {
+                if (!string.IsNullOrWhiteSpace(result.RuleId) || result.RuleIndex >= 0)
+                {
+                    //  {0}: This result specifies both 'result.ruleId' and 'result.rule'.
+                    // Prefer 'result.ruleId' because it is shorter and just as clear.
+                    LogResult(
+                        resultPointer,
+                        nameof(RuleResources.SARIF2004_OptimizeFileSize_Warning_AvoidDuplicativeResultRuleInformation_Text));
+                    return;
+                }
+                else if (result.Rule.ToolComponent == null)
+                {
+                    // {0}: This result uses the 'rule' property to specify the rule
+                    // metadata, but the 'ruleId' property suffices because the rule
+                    // is defined by 'tool.driver'. Prefer 'result.ruleId' because it
+                    // is shorter and just as clear.
+                    LogResult(
+                        resultPointer,
+                        nameof(RuleResources.SARIF2004_OptimizeFileSize_Warning_PreferRuleId_Text));
+                    return;
```

Comment:
>return [](start = 20, length = 6)

There is no need for either of these return statements. There is no loop here. No matter which branch the code takes, control will pass out of the end of the `if` statement and out of this function. I think you were trying to respond to my comment below...

Reply:
updated the wrong place. removing that

---
In reply to: [455344319](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455344319) [](ancestors = 455344319)